sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractConfiguration.java

```diff
@@ -19,67 +19,75 @@
-
 import java.io.IOException;
 import java.nio.file.DirectoryStream;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.stream.StreamSupport;
+import org.sonar.api.batch.ScannerSide;
+import org.sonar.api.config.Settings;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
 
-@BatchSide
 @ScannerSide
 public abstract class AbstractConfiguration {
 
   private static final Logger LOG = Loggers.get(AbstractConfiguration.class);
 
   private final Settings settings;
-  private Boolean reportsComingFromMSBuild;
 
   public AbstractConfiguration(Settings settings) {
     this.settings = settings;
   }
 
-  public abstract String getRoslynJsonReportPathProperty();
+  protected abstract String getRoslynJsonReportPathProperty();
 
-  public abstract String getAnalyzerWorkDirProperty();
+  protected abstract String getAnalyzerWorkDirProperty();
 
-  public abstract String getAnalyzerReportDir();
+  protected abstract String getAnalyzerReportDir();
 
-  public boolean isReportsComingFromMSBuild() {
-    if (reportsComingFromMSBuild == null) {
-      reportsComingFromMSBuild = areProtobufReportsPresent();
-    }
-    return reportsComingFromMSBuild;
-  }
-
-  private boolean areProtobufReportsPresent() {
+  public boolean areProtobufReportsPresent() {
     if (!settings.hasKey(getAnalyzerWorkDirProperty())) {
+      LOG.warn("Property missing: '" + getAnalyzerWorkDirProperty() + "'. No protobuf files will be loaded.");
       return false;
     }
-    Path analyzerOutputDir = protobufReportPathFromScanner();
+    Path analyzerOutputDir = protobufReportPath();
 
     if (!analyzerOutputDir.toFile().exists()) {
-      LOG.info("Analyzer working directory does not exist");
+      LOG.warn("Analyzer working directory does not exist: " + analyzerOutputDir.toAbsolutePath() + ". No protobuf files will be loaded.");
       return false;
     }
 
-    try (DirectoryStream<Path> files = Files.newDirectoryStream(analyzerOutputDir, p -> p.toAbsolutePath().toString().toLowerCase().endsWith(".pb"))) {
+    try (DirectoryStream<Path> files = Files.newDirectoryStream(analyzerOutputDir, p -> p.getFileName().toString().toLowerCase().endsWith(".pb"))) {
       long count = StreamSupport.stream(files.spliterator(), false).count();
+      if (count == 0) {
+        LOG.warn("Analyzer working directory contains no .pb file(s). No protobuf files will be loaded.");
+        return false;
+      }
       LOG.info("Analyzer working directory contains " + count + " .pb file(s)");
-      return count != 0;
+      return true;
     } catch (IOException e) {
-      LOG.warn("Could not check for .pb files in " + analyzerOutputDir.toAbsolutePath().toString(), e);
+      throw new IllegalStateException("Could not check for .pb files in " + analyzerOutputDir.toAbsolutePath(), e);
+    }
+  }
+
+  public boolean isRoslynReportPresent() {
+    if (!settings.hasKey(getRoslynJsonReportPathProperty())) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1126)